ui/types/vault/app-types.ts

```diff
@@ -41,6 +42,23 @@ export interface ModelValidations {
   invalidFormMessage: string;
 }
 
+export interface Model extends Omit<EmberDataModel, 'isNew'> {
+  // override isNew which is a computed prop and ts will complain since it sees it as a function
+  isNew: boolean;
+}
+
+export interface WithFormFieldsModel extends Model {
+  formFields: Array<FormField>;
+  formFieldGroups: FormFieldGroups;
+  allFields: Array<FormField>;
+}
+
+export interface WithValidationsModel extends Model {
+  validate(): ModelValidations;
+}
+
+export interface WithFormFieldsAndValidationsModel extends WithFormFieldsModel, WithValidationsModel {}
+
```

Comment:
It was getting repetitive having to define all of these fields on each model interface so I created some that can be extended based on what decorators are used.